app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueue.java

```diff
@@ -303,6 +304,18 @@ public synchronized void remove(final int index) {
         broadcast(new RemoveEvent(index, getIndex()));
     }
 
+    /**
+     * Removes all the items from the play queue.
+     */
+    public synchronized void removeAll() {
+        while (streams.size() > 0) {
+            final int index = streams.size() - 1;
+            removeInternal(index);
+        }
```

Comment:
To add some reasoning behind it: `removeInternal` is `synchronized` meaning each call has some overhead of synchronizing it. That synchronizations between calls is not needed in this case (i think so at least), since we just want to remove all of them at once

Reply:
done